tests/topotests/bgp_rr_ipv6_shared_ibgp/rr/bgpd.conf

```diff
@@ -0,0 +1,23 @@
+router bgp 65000
```

Comment:
All the peers here are iBGP.  We are seeing the issue when the prefix is advertised towards an eBGP peer, still on the same network.  Because they still share same network, we expect the RR to leave the NH unchanged, which it does, but it also inserts a link-local.

Reply:
In terms of debugging, if we enable BGP update debugging we can see the incoming update without a link-local and then us (RR) sending it with a link-local.

Reply:
For your use-case, you will need "neighbor XXXX nexthop-local unchanged". If unset, it will keep inserting the link-local from the route-reflector. This is what "neighbor XXXX nexthop-local unchanged" is designed for.

Reply:
That also doesn't (or didn't) work to prevent the (F)RR adding in it's link-local.  Although we could live with using it this way, it is different behaviour to IPv4 case where nexthop is unchanged when all iBGP and eBGP neighbours share the same segment without any additional flags.

Reply:
ah I see the proposed change.  Thank you for the solution!